app/models/ont/run_factory.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+# Ont namespace
+module Ont
+  # RunFactory
+  # The factory will create runs from a list of libraries
+  class RunFactory
+    include ActiveModel::Model
+
+    validate :check_run, :check_flowcells
+
+    def initialize(library_names = [])
+      @flowcells = []
+      build_run(library_names)
+    end
+
+    attr_reader :run
+
+    def save(**options)
+      return false unless options[:validate] == false || valid?
+
+      @run.save(validate: false)
+      @flowcells.each { |flowcell| flowcell.save(validate: false) }
+      true
+    end
+
+    private
+
+    attr_reader :flowcells
+
+    def build_run(library_names)
```

Comment:
Which would change the was `idx` is used here as the position, as it is being passed in